src/ArrayParser.php

```diff
@@ -18,7 +18,7 @@ final class ArrayParser
      */
     public function parse(string|null $value): array|null
     {
-        return $value !== null
+        return $value !== null && $value[0] === '{'
```

Comment:
Can value be empty string? In this case will be warning on `$value[0]`.

Reply:
In normal cases it can be `null` or string `{...}`
for default values it can be `null`, `{...}` or an expression
I'm not sure about empty strings it could be for defult values only but I cannot generate that case.